dali/imgcodec/decoders/decoder_parallel_impl.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
I think it would be overdoing things. This implementation decodes samples in parallel. There may be other ways to achiever parallelism (I could imagine a custom decoder scheduling decoding of tiles/stripes in a thread pool).